client/my-sites/plans-grid/components/actions.tsx

```diff
@@ -267,7 +267,9 @@ const LoggedInPlansFeatureActionButton = ( {
 		( addOn ) =>
 			selectedStorageOptionForPlan && addOn?.featureSlugs?.includes( selectedStorageOptionForPlan )
 	)?.checkoutLink;
-	const nonDefaultStorageOptionSelected = defaultStorageOption !== selectedStorageOptionForPlan;
+	const nonDefaultStorageOptionSelected =
+		selectedStorageOptionForPlan !== undefined ||
+		selectedStorageOptionForPlan !== defaultStorageOption;
```

Comment:
I'm also having trouble reproducing the original issue. However, shouldn't this be `selectedStorageOptionForPlan !== undefined && selectedStorageOptionForPlan !== defaultStorageOption`; ? Currently it reads like as long as a storage option is picked and then `nonDefaultStorageOptionSelected` will be true, which doesn't sound like what we intend here :)